### PR TITLE
Allow function with binary-ish expression to hug

### LINKED ISSUE #2446: Inconsistent wrapping of body of implicit-return arrow functions
If I use an arrow function that returns a single expression, it wraps as I'd expect, with the params on the initial line and the body indented (similar to how non-implicit-return arrow functions wrap. (Both examples show the post-processed code—see sandbox link below.)

```js
// This wraps as I'd expect
array.filter((element, index) =>
  element.field.some(someVeryLongPredicateFunction)
);
```

But if the function body is some kind of operator expression, it wraps the params as well

```js
// This wraps strangely
array.filter(
  (el, index) => somePredicate(index) && el.field.some(someLongPredicate)
);
```

This seems strangely inconsistent. Is this intentional?

[See sandbox](https://prettier.io/playground/#%7B%22content%22%3A%22%2F%2F%20This%20wraps%20as%20I'd%20expect%20%2F%20write%20myself%5Cnarray.filter((element%2C%20index)%20%3D%3E%5Cn%20%20element.field.some(someVeryLongPredicateFunction)%5Cn)%5Cn%5Cn%2F%2F%20This%20wraps%20strangely%5Cnarray.filter((element%2C%20index)%20%3D%3E%20%5Cn%20%20somePredicate(index)%20%26%26%20element.field.some(someVeryLongPredicate)%5Cn)%5Cn%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22babylon%22%2C%22semi%22%3Atrue%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%2C%22ast%22%3Afalse%7D%7D)

### PR DESCRIPTION
**Pull request by azz**
This one might be a little controversial.

Fixes #2446

### CODE REVIEW

File: tests/binary-expressions/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -8,12 +8,11 @@ function f() {
 }
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 function f() {
-  const appEntitys = getAppEntitys(loadObject).filter(
-    entity =>
-      entity &&
-      entity.isInstallAvailable() &&
-      !entity.isQueue() &&
-      entity.isDisabled()
+  const appEntitys = getAppEntitys(loadObject).filter(entity =>
```
Comment (azz) [inline tests/binary-expressions/__snapshots__/jsfmt.spec.js.snap]: I personally prefer the old behaviour.

I tried briefly to only apply the group `ifBreak`, but I'm not sure that's possible (`willBreak` is insufficient here).

**Comment by vjeux** [comment]
It seems like it's not a clear win across the board so we should probably not do it. It's just going to make codebase of people change and make it harder to upgrade. What do you think?

**Comment by azz** [comment]
Yeah I'm not sure either way. On one hand it is more consistent in general, on the other hand it produces some code that I wouldn't write myself. 🤷‍♂️

**Comment by azz** [comment]
Going to close this one. Probably not the right move, we can always revisit if there's a better solution.

**Pull request closed without merging**